transport-native-unix-common/src/main/java/io/netty/channel/unix/Unix.java

```diff
@@ -53,11 +53,9 @@
      * @param registerTask
      */
     @UnstableApi
-    public static void registerInternal(Runnable registerTask) {
-        if (registered.compareAndSet(false, true)) {
-            registerTask.run();
-            Socket.initialize();
-        }
+    public static synchronized void registerInternal(Runnable registerTask) {
+        registerTask.run();
+        Socket.initialize0(true);
```

Comment:
Could we alternatively leave `Socket` unchanged, and then here do something like this:

```
if (registered.compareAndSet(false, true)) {
             registerTask.run();
}
Socket.initialize();
```

Since `Socket.initialize();` does it's own "init only once" check, there's no need to guard it.

Reply:
The way it is here, if you call `Socket.initialize();` and then call `Unix.registerInternal(…);`, then socket will be initialised twice.

Reply:
Yeah and that is exactly what we need to do as `Socket.initialize` will call a JNI method which will init static fields. As every of our native libs statically compile the unix-common stuff in we will get strange side-effects otherwise

Reply:
interesting idea... Let me poke a bit in the code

Reply:
was done in an extra PR